hpx/runtime/serialization/basic_archive.hpp

```diff
@@ -57,7 +57,7 @@ namespace hpx { namespace serialization
         {}
 
         template <typename T>
-        void invoke(T * t)
+        void invoke(T * (&t))
         {
```

Comment:
This change regresses functionality as it now rejects cv-qualified pointers for the pointer overload. It is a subtle change in an already subtle mechanism. Given that the pointer overload does nothing but `static_assert`, I suggest dropping it as an overload and doing the appropriate `static_assert`ing in the main function instead.